modules/media.js

```diff
@@ -218,36 +264,41 @@ Media.prototype.get = function(type, directory, forEach) {
     directory = null;
   }
 
-  if (!window.ffosbr.utils.isFunction(forEach)) {
+  if (!ffosbr.utils.isFunction(forEach)) {
     throw new Error('Missing or invalid callback');
   }
 
   storages = this.getStorageByName(type === 'sdcard1' ? 'sdcard' : type);
+
   internal = storages.internal;
   external = storages.external;
 
-  if (type === 'sdcard1' && external.ready === true) {
-    if (directory !== null) {
-      externalFiles = external.enumerate(directory);
-    }
-  } else if (type === 'sdcard1' && external.ready === false) {
-    throw new Error('Attempt to read from an invalid storage. Abort.');
-  } else if (internal.ready === true || external.ready === true) {
+  if (type === 'sdcard1') {
+    externalFiles = external.store.enumerate(directory);
```

Comment:
I think passing null would actually cause issues, but in this case (I believe, so correct me if I'm wrong) that directory can only have one of two types: string or undefined. If directory is a string, then it will interpreted as a file path by enumerate and everything's fine. If directory is undefined, it's the same as not passing a parameter, and this is also fine.